### ISSUE TITLE
Repeat within entities

### ISSUE THREAD

**Issue by wonderchook** [Cadasta/cadasta-platform#1452]
collect multiple attributes of a field within a single entity 

https://devwiki.corp.cadasta.org/XLS%20repeat%20feature

**Comment by dpalomino** [comment]
I've updated the original description to include the requirements. 

The implementation for ODK/GeoODK means to generate the xform as defined in the [reference page](http://xlsform.org/#repeats). I think that is the relatively easy part. 

However, to be able to display this in a proper way in the WUI could be trickier (@clash99, thoughts?) . And I think we need to evaluate if our current database model can support this feature (@amplifi @oliverroick, what do you think?). 

In any case, I think this will come not before the refactoring of jsonattrs. 

Any feedback will be welcome!

**Comment by wonderchook** [comment]
@dpalomino I think we can support it, but I'm a bit worried this is going to be difficult for partners to use after they have collected the data. The further we stray away from essentially single table Excel type data the harder it is going to be for people to understand.

**Comment by dpalomino** [comment]
> @dpalomino I think we can support it, but I'm a bit worried this is going to be difficult for partners to use after they have collected the data. The further we stray away from essentially single table Excel type data the harder it is going to be for people to understand.

Yeah, that's true. I think it will be easier when we will develop the Data Designer (xform generator). Meanwhile at least it'd be useful for us when creating some of the questionnaires for partners. Not a top priority though but something for the medium term...

**Comment by dpalomino** [comment]
Updating to the new format: 

### Problem Statement

Partners are collecting repeatable groups of information within an entity, so the support of repeat within a location, relationship or party entity would be useful.

### Context / Use Case

Examples of questionnaires that would benefit of supporting repeats within entities: 
- Including a list of witnesses 
- List of crops/fertiliser/herbicides in a farm
- List of legal documents with their description
- etc

### User Stories

- [ ] US1: As a data collector using ODK/GeoODK, I want to collect multiple repeats groups within an entity (either a location, tenure_relationship or party)
- [ ] US2: As a data collector using the web UI, I want to collect multiple repeats groups within an entity (either a location, tenure_relationship or party)
- [ ] US3: As a project user using the web UI, I want to see the the information contained in the "repeated groups" within each entity 

### Description 

#### Supporting `begin repeat` and `end repeat` within entities in xforms 

This is well defined in the [xlsform reference page](http://xlsform.org/#repeats). 

I think an example will be much more descriptive: 

```
begin group    location_attributes
 text          location_name        Name of the location
 begin repeat 
     text      cultivation_detail    What are you cultivating in your land?
     ...
 end repeat 
end group 
```
```
begin group    party_attributes_individual
 date          dob                  Date of birth
 begin repeat 
     text      relative_name    Name of the relative living with you
     ...
 end repeat 
end group 
```

#### Supporting nested repeat groups in the web UI

- We need to define some wireframes to support these nested repeat groups for each entity 
- As a note, we need to check how feasible/difficult would be to support this in our current databas model
